src/storage/txn/actions/acquire_pessimistic_lock.rs

```diff
@@ -82,148 +85,41 @@ pub fn acquire_pessimistic_lock<S: Snapshot>(
-            if let Some((write, commit_ts)) = write {
-                // Here `get_write_with_commit_ts` returns only the latest PUT if it exists and
-                // is not deleted. It's still ok to pass it into `check_data_constraint`.
-                check_data_constraint(reader, should_not_exist, &write, commit_ts, &key).or_else(
-                    |e| {
-                        if is_already_exist(&e) && commit_ts > requested_for_update_ts {
-                            // If `allow_lock_with_conflict` is set and there is write conflict,
-                            // and the constraint check doesn't pass on the latest version,
-                            // return a WriteConflict error instead of AlreadyExist, to inform the
-                            // client to retry.
-                            // Note the conflict_info may be not consistent with the
-                            // `locked_with_conflict_ts` we got before.
-                            // This is possible if the key is locked by a newer request with
-                            // larger for_update_ts, in which case the result of this request
-                            // doesn't matter at all. So we don't need
-                            // to care about it.
-                            let conflict_info = ConflictInfo {
-                                conflict_start_ts: write.start_ts,
-                                conflict_commit_ts: commit_ts,
-                            };
-                            return Err(conflict_info.into_write_conflict_error(
-                                reader.start_ts,
-                                primary.to_vec(),
-                                key.to_raw()?,
-                            ));
-                        }
-                        Err(e)
-                    },
-                )?;
-
-                if need_load_value {
-                    val = Some(reader.load_data(&key, write)?);
-                } else if need_check_existence {
-                    val = Some(vec![]);
-                }
-            }
-        }
-        // Previous write is not loaded.
-        let (prev_write_loaded, prev_write) = (false, None);
-        let old_value = load_old_value(
-            need_old_value,
-            need_load_value,
-            val.as_ref(),
-            reader,
-            &key,
-            for_update_ts,
-            prev_write_loaded,
-            prev_write,
-        )?;
-
-        // Overwrite the lock with small for_update_ts
-        if for_update_ts > lock.for_update_ts {
-            let lock = PessimisticLock {
-                primary: primary.into(),
-                start_ts: reader.start_ts,
-                ttl: lock_ttl,
-                for_update_ts,
+    let mut current_lock = reader.load_lock(&key)?;
+    if let Some(lock) = current_lock.as_ref() {
+        if is_shared_lock_req && lock.is_shared() && lock.contains_start_ts(reader.start_ts) {
```

Comment:
Is it possbile to miss some branches here?

How about seprating the conditions so there would be no missing, like: 
```
If is_shared_lock_req {
  if lock.is_shared() {
  } else {
  }
} else {
   let is_same_lock = req.start_ts == lock.start_ts;
   if lock.is_shared() {
  } else {
  }
}
```